### DISCUSSION TITLE
FUNÇÕES DELEGATE

### DISCUSSION THREAD

**Discussion by pedrovazfontes** [balta-io/forum#225] [Tira Dúvidas] [answered]
boa tarde Estou com um problema, estou realizando um "Crud". Criei funções para cadastrar o fornecedor, uma função que vem através do delegate na classe Fornecedor. O problema é que quando vou dar um

Console.WriteLine($"{cad_fornecedor.Nome})

ele não encontra as propriedades da classe Fornecedor:
using System;

namespace Package{
    class Program
    {   
        public static void Main(string[] args)
        {
            var opcao = "";
            var cad_fornecedor = new Fornecedor.CadastrarFornecedor(NovoFornecedor);
            var cad_produto = new Produtos.CadastrarProdutos(NovoProduto);
        
            while(opcao != "4"){
                Console.WriteLine("\n[1] Cadastrar Fornecedor \n" +
                "[2] Cadastrar Produto \n" +
                "[3] Realizar Venda \n" +
                "[4] Sair \n" );
                Console.Write("De acordo com as opções abaixo, digite a desejada: ");
                opcao = Console.ReadLine();

                if(opcao == "1"){
                    cad_fornecedor();
                    IList<Fornecedor> cadastros = new List<Fornecedor>();
                    
                    
                }

                else if(opcao == "2"){
                    NovoProduto();
                    Console.WriteLine($"{cad_fornecedor.}");
                }

                else if(opcao == "3"){
                    Console.WriteLine("ok!");
                }
            }
            
  
        }

        public static void NovoFornecedor(){
            var fornecedor = new Fornecedor();
            Console.Write("Digite o nome do Fornecedor: ");
            fornecedor.Nome = Console.ReadLine();
            Console.Write("Agora digite o CNPJ: ");
            fornecedor.CNPJ = Console.ReadLine();
            Console.Write("Agora digite a data do Inicio do Contrato(dd/mm/yyyy): ");
            fornecedor.InicioContrato = (Console.ReadLine());
            Console.Write("Ok, o contrato ainda está ativo? ");
            fornecedor.VerificaContrato();
            Console.WriteLine("Abaixo o resultado das informações");
     
        }

        public static void NovoProduto(){
            var produto = new Produtos();
            Console.Write("Digite o nome do Produto: ");

        }
    
        
    }

   
} 

namespace Package
{

    public class Fornecedor
    {
        public string? Nome;
        public string? CNPJ ;
        public string? InicioContrato;
        public bool? _contratoAtivo ;
       
        public void VerificaContrato(){

            Console.Write("Seu contrato está ativo? ");
            var pergunta = Console.ReadLine();
            if (pergunta != "sim"){
                Console.WriteLine("Ok, acionar setor de Compras e verificar situação!");
                _contratoAtivo = false;
            } else{
                Console.WriteLine("Perfeito!");
                _contratoAtivo = true;
            }
       }
       
       public delegate void CadastrarFornecedor();
       
    }
}

![print](https://user-images.githubusercontent.com/61070351/212564734-98b83279-d14e-41b7-97f1-f21d338e16f2.png)

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @pedrovazfontes, qual curso/aula você está?
Também não consegui visualizar qual erro ocorreu, poderia dar mais detalhes por favor?

**Comment by pedrovazfontes** [comment]
Bom dia
Estou na aula de Delegates
No caso o problema é que não consigo acessar o valor da propriedade Nome em um Console.WriteLine

**Reply by andrebaltieri** [reply]
Bom dia, @pedrovazfontes, pelo que vi ali no seu código, o cursor está depois das aspas fechadas. Teria que ser `cad_fornecedor.Nome` tudo junto.
